src/main/java/com/minecolonies/coremod/network/messages/SchematicSaveMessage.java

```diff
@@ -0,0 +1,117 @@
+package com.minecolonies.coremod.network.messages;
+
+import com.minecolonies.coremod.configuration.Configurations;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.Structures;
+import com.minecolonies.coremod.util.*;
+import com.minecolonies.structures.helpers.Structure;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler;
+import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Save Schematic Message.
+ */
+public class SchematicSaveMessage implements IMessage, IMessageHandler<SchematicSaveMessage, IMessage>
+{
+    private       byte[] data           = null;
+    final private int    MAX_TOTAL_SIZE = 32767;
+
+    /**
+     * Public standard constructor.
+     */
+    public SchematicSaveMessage()
+    {
+        super();
+    }
+
+    /**
+     * Send a schematic to the client.
+     *
+     * @param data byte array of the schematic.
+     */
+    public SchematicSaveMessage(final byte[] data)
+    {
+        this.data = data;
+    }
+
+    @Override
+    public void fromBytes(@NotNull final ByteBuf buf)
+    {
+        int length = buf.readInt();
+        final byte[] compressedData = new byte[length];
+        buf.readBytes(compressedData);
+        data = Structure.uncompress(compressedData);
+    }
+
+    @Override
+    public void toBytes(@NotNull final ByteBuf buf)
+    {
+        final byte[] compressedData = Structure.compress(data);
```

Comment:
We could make an OP check here and allow OPS (admins) to load up schematics even if turned off in the configs.
What do you think?